pkg/cloud/aws/actuators/scope.go

```diff
@@ -164,22 +170,32 @@ func (s *Scope) Close() {
 		return
 	}
 
-	pb, err := json.MarshalIndent(p, "", "  ")
-	if err != nil {
-		s.Error(err, "failed to json marshal patch")
-		return
+	// Do not update Machine if nothing has changed
+	if len(p) != 0 {
+		pb, err := json.MarshalIndent(p, "", "  ")
+		if err != nil {
+			s.Error(err, "failed to json marshal patch")
+			return
+		}
+		s.Logger.V(1).Info("Patching cluster")
+		result, err := s.ClusterClient.Patch(s.Cluster.Name, types.JSONPatchType, pb)
+		if err != nil {
+			s.Error(err, "failed to patch cluster")
+			return
+		}
+		// Keep the resource version updated so the status update can succeed
+		s.Cluster.ResourceVersion = result.ResourceVersion
```

Comment:
This seems a bit confusing to just update the resource version, would it be better to update the entire `Machine` object here, or possibly the returned object as a separate variable before doing the status update?

Reply:
This is the minimal amount of updating necessary to make sure the UpdateStatus call succeeds. 

The code gets weird if I'm saving to an alternative variable, it looks like this:

```
var updatedCluster *v1alpha1.Cluster
if len(p) > 0 {
   ... run updates
   updatedCluster, err = Patch()
}

if updatedCluster != nil  {
   if reflect.DeepEqual(updatedCluster, s.ClusterStatus) {
    return
  }
}
 UpdateStatus(updatedClusteR)

```

What is the benefit of updating the entire machine? Especially given we're just on the verge of chucking the whole thing away as the scope closes

Reply:
My main concern is around future changes that might break some of the assumptions we have in place, but I'm probably just being overly paranoid.